### DISCUSSION TITLE
How to make xll file with new project following docs sample

### DISCUSSION THREAD

**Discussion by theangkko** [cunnane/xloil#60] [Q&A]
I am following to make XLL sample. I tried to do something, but I have failed ( cause I am a newcomer in VC++.)
Please help what I need to fix.

I want to do below section from DOC.
![image](https://github.com/cunnane/xloil/assets/75212211/e1dea9ec-6283-4596-9e95-4809f3f46c08)

<br/><br/>

But, I fails to resolve from include error.
Even though I compare between "TestAddin" which is mentioned Alternative method and "my-NewTestAddin" as new project, I cannot catch what I have to do.

I have old module using Excel-Dna(c#), So I thought I could make xll module.  <br> It is not easy to make xll of c++ version.  

For your review, I add my test solution.
https://github.com/theangkko/xlOil_try

![image](https://github.com/cunnane/xloil/assets/75212211/8071b788-c456-451a-9bfb-2bee64e1ed4b)

<img width="1329" alt="image" src="https://github.com/cunnane/xloil/assets/75212211/39d2af47-7720-402d-9308-ee97bcc2e8f6">

_

**Comment by cunnane** [comment]
The xlOil.h header file is autogenerated during the release build and staging process.  I've realised this isn't very convenient for building static XLLs, so I have changed this to be part of the TestAddin build (this change is in the tip of the 0.17 branch).  This should allow you to duplicate the TestAddin project and get an XLL build working.  

I will also make a change to the build scripts (which currently build all the python binaries) to make it easier to stage the only the C++ binaries.

**Reply by theangkko** [reply]
Thanks for your help. I will to try and leave a feedback again. 
(I am so sorry that I have not visited due to movement!)

**Reply by theangkko** [reply]
I have tried after getting your reply. But I have failed to make xll (build OR rebuild with right click on the test project)

Am I right to follow right procedure to make XLL project? I understand you fix some file to make eaiser xll project when I wrote query. After then whenever I tried 4~5 times, I got similar error.

If I uploaded log and captures, can you help?

**Reply by cunnane** [reply]
I've just pushed some build fixes to v0.18 and main branches - does that help?  What issues are you seeing?

**Reply by theangkko** [reply]
I have failed still now, I will try again and capture those process. 
Anyway! thanks for your help!
